src/Form/ImportantSolrSettingsForm.php

```diff
@@ -0,0 +1,249 @@
+    $response->addCommand(new ReplaceCommand("#solr-field-select", ($form['type']['solr_field_select'])));
+    
+    return $response;
+   }
+
+  /**
+   * AJAX callback function when user selects Index
+   *
+   * Updates Solr Field dropdown
+   */
+  public function onIndexSelect(array &$form, FormStateInterface $form_state) {
+    // Get solr fields for selected Index
+    $fields = $this->getSolrFields($form_state->getValue('index_select'));
+    $form['type']['solr_field_select']['#options'] = $fields;
+    
+    return $form['type']['solr_field_select'];  
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('strawberryfield.archipelago_solr_settings');
+    
+    // Get all Solr servers
+    $servers = $this->entityTypeManager
+      ->getStorage('search_api_server')
+      ->loadMultiple();
+
+    // Create a fieldset for the Type setting
+    $form['type'] = array(
+      '#type' => 'fieldset',
+      '#title' => $this->t('Solr Field for Type in View Mode mapping'),
+      '#markup' => '<p>Choose the Solr field that will be used in our ADO to View Mode mapper. <br><br> Fields are specific to their Solr Server and Index. <br> To edit servers, indexes, and fields, visit <a href=" /admin/config/search/search-api/">Search API Config</a><br> For info and settings about the View Mode mapper, visit the <a href="/admin/config/archipelago/viewmode_mapping">ADO to View Mode Config</a></p>'
+    );
+
+    $form['type']['server_select'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Step 1. Select Server'),
+      '#options' => $this->formatOptions($servers),
+      '#default_value' => $config->get('type_server'),
+      '#ajax' => [
+        'callback' => '::onServerSelect',
+        'disable-refocus' => FALSE,
+        'event' => 'change',
+        'wrapper' => 'index-select', 
+      ]
+    ];
+    
+    $form['type']['index_select'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Step 2. Select Solr Index'),
+      '#prefix' => '<div id="index-select">',
+      '#suffix' => '</div>',
+      '#options' => $config->get('type_server') ? $this->getIndexes($config->get('type_server')) : null,
+      "#empty_value" => "",
+      "#default_value" => $config->get('type_index'),
+      // If not #validated, dynamically populated dropdowns don't work.
+      '#validated' => TRUE,
+      '#ajax' => [
+        'callback' => '::onIndexSelect',
+        'disable-refocus' => FALSE, 
+        'event' => 'change',
+        'wrapper' => 'solr-field-select', 
+      ]
+    ];
+    
+    $form['type']['solr_field_select'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Step 3. Select Solr Field'),
+      '#prefix' => '<div id="solr-field-select">',
+      '#suffix' => '</div>',
+      '#options' => $config->get('type_server') && $config->get('type_index') ? $this->getSolrFields($config->get('type_index')) : null,
```

Comment:
We need better logic here, and that can go into its own function:
The Solr field should have a set of certain conditions to be usable (selectable) in this context.

```YAML
 type_1:
    label: 'Descriptive Metadata » type'
    datasource_id: 'entity:node'
    property_path: 'field_descriptive_metadata:type'
    type: string
    dependencies:
      config:
        - field.storage.node.field_descriptive_metadata
```

So in specific, fields that can be used are the ones where their data sources is a node and the property path contains a field that is of type SBF. I know this adds some overhead to your code, but it you can add/reuse some of the functionality we already have in this module, you will notice its not that complex. And we can reuse that logic quite a lot if you generalize it (can go, e.g into the utils service we provide, just saying).
Always check existing configurations so you can see how things are being done (and what data you have available to make smarter selections/filters), e.g the Search Index config YAML
![single export](https://user-images.githubusercontent.com/6946023/71989702-1f9d1b00-3200-11ea-9008-3c360689bc0b.jpg)